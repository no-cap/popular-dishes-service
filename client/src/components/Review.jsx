import React from 'react';
import StarBox from './starbox.jsx';
import Star from './star.jsx';
import PersonIcon from './personIcon.jsx';
import StarType from '../components-style/TypesOfStars.jsx';

import { 
    SingleReview, 
    PhotoCropper, 
    ProfilePic, 
    UserName, 
    TinyPairIcon, 
    FriendCount, 
    TinyStar, 
    UserReviews,
    StarContainer,
    Date,
    ReviewText
} from '../components-style/SingleReview.jsx'

class Review extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text: this.props.data.text,
            readMore: false
        };
        this.onClickHandler = this.onClickHandler.bind(this);
        this.dateFormatter = this.dateFormatter.bind(this);
    }
    // handles clicking on Read more / Read less
    onClickHandler(e) {
        this.setState({ readMore: !this.state.readMore })
    }
    // Formats Dates
    dateFormatter(date) {
        var arr = date.slice(0, 11).split("-");
        if (arr[1][0] === '0') {
            var dateWithZero = arr[1] + '/' + arr[0] + '/' + arr[2];
            return dateWithZero.slice(1);
        } else {
            return arr[1] + '/' + arr[0] + '/' + arr[2];
        }
    }

    render() {
        return (
            <SingleReview >
                <PhotoCropper> <ProfilePic src={this.props.data.userphoto} /> </PhotoCropper>
                <UserName> {this.props.data.username} </UserName>
                <TinyPairIcon > 
                    <PersonIcon /> 
                </TinyPairIcon>
                <FriendCount > {this.props.data.friends} </FriendCount>
                <TinyStar> 
                    <StarType type="mini" rating={4}> <Star/> 
                </StarType> </TinyStar>
                <UserReviews> {this.props.data.reviews} </UserReviews>
                <StarContainer> 
                    <StarBox rating={this.props.data.rating} /> 
                </StarContainer>
                <Date> {this.dateFormatter(this.props.data.date)} </Date>
                <ReviewText>
                    <div id={this.props.data.user}>{(this.state.readMore) ? this.state.text : this.state.text.split(" ").slice(0, 22).join(" ")}</div>
                    {(this.props.data.text.split(" ").length > 22) ? <button onClick={this.onClickHandler}> {(this.state.readMore) ? 'Read less' : 'Read more'} </button> : null}
                </ReviewText>
            </SingleReview>
        )
    }
}

export default Review;


// { JSON.stringify(this.props.data) }