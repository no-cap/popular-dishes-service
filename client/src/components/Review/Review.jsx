import React from 'react';
import StarBox from '../Starbox/Starbox';
import Star from '../Starbox/Star';
import PersonIcon from '../PersonIcon/PersonIcon';
import StarType from '../Starbox/StarStyles';

import {
  SingleReview,
  PhotoCropper,
  ProfilePic,
  UserName,
  TinyPairIcon,
  FriendCount,
  TinyStar,
  UserReviews,
  StarContainer,
  Date,
  ReviewText,
} from './ReviewStyles';

class Review extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      text: this.props.data.text,
      readMore: false,
    };
    this.onClickHandler = this.onClickHandler.bind(this);
    this.dateFormatter = this.dateFormatter.bind(this);
  }

  // handles clicking on Read more / Read less
  onClickHandler(e) {
    this.setState({ readMore: !this.state.readMore });
  }

  // Formats Dates
  // eslint-disable-next-line class-methods-use-this
  dateFormatter(date) {
    const arr = date.slice(0, 11).split('-');
    if (arr[1][0] === '0') {
      const dateWithZero = `${arr[1]}/${arr[0]}/${arr[2]}`;
      return dateWithZero.slice(1);
    }
    return `${arr[1]}/${arr[0]}/${arr[2]}`;
  }

  render() {
    return (
      <SingleReview>
        <PhotoCropper>
          <ProfilePic src={this.props.data.userphoto} />
        </PhotoCropper>
        <UserName>{this.props.data.username}</UserName>
        <TinyPairIcon>
          <PersonIcon />
        </TinyPairIcon>
        <FriendCount>{this.props.data.friends}</FriendCount>
        <TinyStar>
          <StarType type="mini" rating={4}><Star /></StarType>
        </TinyStar>
        <UserReviews>{this.props.data.reviews}</UserReviews>
        <StarContainer>
          <StarBox rating={this.props.data.rating} />
        </StarContainer>
        <Date>{this.dateFormatter(this.props.data.date)}</Date>
        <ReviewText>
          <div id={this.props.data.user}>{(this.state.readMore) ? this.state.text : this.state.text.split(' ').slice(0, 22).join(' ')}</div>
          {(this.props.data.text.split(' ').length > 22) ? (
            <button onClick={this.onClickHandler}>{(this.state.readMore) ? 'Read less' : 'Read more'}</button>
          ) : null}
        </ReviewText>
      </SingleReview>
    );
  }
}

export default Review;
