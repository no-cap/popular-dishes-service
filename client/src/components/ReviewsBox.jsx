import React from 'react';
import Review from './Review.jsx';
import { ReviewsLength } from '../components-style/ReviewsBox-style.jsx';

const ReviewsBox = ({ reviews }) => (
    <div>
    <ReviewsLength>Reviews ({reviews.length})</ReviewsLength>
    {reviews.map((review) => <Review data={review}/>)}
    </div>
);

export default ReviewsBox;