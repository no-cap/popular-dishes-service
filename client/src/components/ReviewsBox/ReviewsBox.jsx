import React from 'react';
import Review from '../Review/Review';
import { ReviewsLength } from './ReviewsBoxStyle';

const ReviewsBox = ({ reviews }) => (
  <div>
    <ReviewsLength>{`Reviews (${reviews.length})`}</ReviewsLength>
    {reviews.map((review) => <Review key={review.reviewId} review={review} />)}
  </div>
);

export default ReviewsBox;
