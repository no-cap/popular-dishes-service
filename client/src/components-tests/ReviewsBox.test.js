import React from 'react';
import { shallow,mount } from 'enzyme';
import ReviewsBox from '../components/ReviewsBox.jsx';
import Review from '../components/Review.jsx';


var sampleData = [
    {
        user: 'Bob',
        text: 'I love corn cheese'
    },
    {
        user: 'Bob',
        text: 'I love corn cheese'
    },
    {
        user: 'Bob',
        text: 'I love corn cheese'
    }
];

describe('ReviewsBox Unit Tests', () => {

    describe('ReviewBox should render correctly', () => {
        test('ReviewsBox should render correctly in "debug" mode', () => {

            const wrapper = shallow(<ReviewsBox reviews={sampleData} />);

            expect(wrapper).toMatchSnapshot();
        });

        test('ReviewsBox should have three children', () => {

            const wrapper = mount(<ReviewsBox reviews={sampleData} />);

            expect(wrapper.find('button')).toHaveLength(10);
        });
    });

});

