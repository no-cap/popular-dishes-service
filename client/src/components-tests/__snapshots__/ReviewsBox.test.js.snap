// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsBox Unit Tests ReviewBox should render correctly ReviewsBox should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
